frontends/concrete-python/concrete/fhe/tfhers/dtypes.py

```diff
@@ -3,32 +3,142 @@
+        carry_modulus: int,
+        max_noise_level: int,
+        log2_p_fail: float,
+        ciphertext_modulus: Optional[int],
+        big_encryption_key: bool,
+    ):
+        self.lwe_dimension = lwe_dimension
+        self.glwe_dimension = glwe_dimension
+        self.polynomial_size = polynomial_size
+        self.lwe_noise_distribution_std_dev = lwe_noise_distribution_std_dev
+        self.glwe_noise_distribution_std_dev = glwe_noise_distribution_std_dev
+        self.pbs_base_log = pbs_base_log
+        self.pbs_level = pbs_level
+        self.ks_base_log = ks_base_log
+        self.ks_level = ks_level
+        self.message_modulus = message_modulus
+        self.carry_modulus = carry_modulus
+        self.max_noise_level = max_noise_level
+        self.log2_p_fail = log2_p_fail
+        self.ciphertext_modulus = ciphertext_modulus
+        self.big_encryption_key = big_encryption_key
+
+    def __str__(self) -> str:
+        return (
+            f"tfhers_params<{self.lwe_dimension}, {self.glwe_dimension}, {self.polynomial_size}, "
+            f"{self.lwe_noise_distribution_std_dev}, {self.glwe_noise_distribution_std_dev}, "
+            f"{self.pbs_base_log}, {self.pbs_level}, {self.ks_base_log}, {self.ks_level}, "
+            f"{self.message_modulus}, {self.carry_modulus}, {self.max_noise_level}, "
+            f"{self.log2_p_fail}, {self.ciphertext_modulus}, {self.big_encryption_key}>"
+        )
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            isinstance(other, self.__class__)
+            and self.lwe_dimension == other.lwe_dimension
+            and self.glwe_dimension == other.glwe_dimension
+            and self.polynomial_size == other.polynomial_size
+            and self.lwe_noise_distribution_std_dev == other.lwe_noise_distribution_std_dev
+            and self.glwe_noise_distribution_std_dev == other.glwe_noise_distribution_std_dev
+            and self.pbs_base_log == other.pbs_base_log
+            and self.pbs_level == other.pbs_level
+            and self.ks_base_log == other.ks_base_log
+            and self.ks_level == other.ks_level
+            and self.message_modulus == other.message_modulus
+            and self.carry_modulus == other.carry_modulus
+            and self.max_noise_level == other.max_noise_level
+            and self.log2_p_fail == other.log2_p_fail
+            and self.ciphertext_modulus == other.ciphertext_modulus
+            and self.big_encryption_key == other.big_encryption_key
+        )
+
+    def __hash__(self) -> int:
+        return hash((self.lwe_dimension, self.glwe_dimension, self.polynomial_size))
+
+
 class TFHERSIntegerType(Integer):
     """
     TFHERSIntegerType (Subclass of Integer) to represent tfhers integer types.
     """
 
     carry_width: int
     msg_width: int
-
-    def __init__(self, is_signed: bool, bit_width: int, carry_width: int, msg_width: int):
+    params: TFHERSParams
+
+    def __init__(
+        self,
```

Comment:
Here it could a TFHERSIntergerType will be composed of crypto params and radix encoding parameter,